src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -142,40 +156,40 @@
     (comidi/GET ["/resource_types/" [#".*" :rest]] request
       (master-request-handler request))
     (comidi/GET ["/status/" [#".*" :rest]] request
-      (master-request-handler request))
+      (master-request-handler request))))
 
-    ;; legacy CA routes
+(defn legacy-ca-routes
+  [ca-request-handler]
+  (comidi/routes
     (comidi/ANY ["/certificate_status/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/ANY ["/certificate_statuses/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_revocation_list/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/PUT ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))))
-
-(defn legacy-routes
-  "Creates all of the comidi routes for the master."
-  [master-request-handler ca-request-handler]
-  (comidi/routes
-    (comidi/context "/v2.0"
-      (v2_0-routes master-request-handler))
-    (comidi/context ["/" :environment]
-      (legacy-master-routes master-request-handler ca-request-handler))))
+                (ca-request-handler request))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(defn build-ring-handler
-  [master-handler master-mount master-api-version
-   ca-handler ca-mount ca-api-version]
-  (comidi/routes->handler
-    (legacy-routes
-      #(request-compatibility-wrapper
-        master-handler master-mount master-api-version %)
-      #(request-compatibility-wrapper
-        ca-handler ca-mount ca-api-version %))))
+(schema/defn ^:always-validate
+  build-ring-handler :- IFn
```

Comment:
As a non-folder, I actually like having the function name on the second line so my eyeballs don't need to skip past the annotation. This is also how the rest of the codebase is written. 

All that said, honestly, I really don't like having this annotation here at all. In personal projects I usually only turn schema checking on during testing, and just leave it off (the default) in prod. But alas, this ain't my project (TAMP).